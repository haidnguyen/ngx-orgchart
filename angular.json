{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "chart": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "inlineStyle": true,
          "inlineTemplate": true,
          "changeDetection": "OnPush",
          "viewEncapsulation": "None",
          "prefix": "app",
          "skipTests": true
        },
        "@schematics/angular:service": {
          "skipTests": true
        },
        "@schematics/angular:directive": {
          "skipTests": true
        },
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/chart",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "chart:build:production"
            },
            "development": {
              "browserTarget": "chart:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "chart:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "src/**/*.ts",
              "src/**/*.html"
            ]
          }
        }
      }
    },
    "ngx-orgchart": {
      "projectType": "library",
      "root": "projects/ngx-orgchart",
      "sourceRoot": "projects/ngx-orgchart/src",
      "prefix": "ngx",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "inlineStyle": true,
          "inlineTemplate": true,
          "changeDetection": "OnPush",
          "viewEncapsulation": "None",
          "prefix": "app",
          "skipTests": true
        },
        "@schematics/angular:service": {
          "skipTests": true
        },
        "@schematics/angular:directive": {
          "skipTests": true
        },
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/ngx-orgchart/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/ngx-orgchart/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/ngx-orgchart/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/ngx-orgchart/**/*.ts",
              "projects/ngx-orgchart/**/*.html"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "chart",
  "cli": {
    "defaultCollection": "@angular-eslint/schematics"
  }
}
