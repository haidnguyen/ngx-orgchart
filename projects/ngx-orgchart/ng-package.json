{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/ngx-orgchart",
  "lib": {
    "entryFile": "src/public-api.ts"
  },
  "allowedNonPeerDependencies": [
    "immutability-helper"
  ]
}
