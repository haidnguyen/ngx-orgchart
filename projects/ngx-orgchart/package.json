{
  "name": "ngx-orgchart",
  "version": "0.0.1",
  "peerDependencies": {
    "@angular/common": "^12.1.1",
    "@angular/core": "^12.1.1",
    "@ngrx/component-store": "^12.2.0"
  },
  "dependencies": {
    "tslib": "^2.2.0",
    "immutability-helper": "^3.1.1"
  }
}
