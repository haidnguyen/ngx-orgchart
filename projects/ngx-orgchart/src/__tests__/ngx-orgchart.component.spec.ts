import { NO_ERRORS_SCHEMA } from '@angular/core';
import { render } from '@testing-library/angular';

import { NgxOrgchartComponent } from '../lib/ngx-orgchart.component';

describe(NgxOrgchartComponent.name, () => {
  it('should match snapshot', async () => {
    const { container } = await render(NgxOrgchartComponent, {
      schemas: [NO_ERRORS_SCHEMA],
    });

    expect(container).toMatchSnapshot();
  });
});
