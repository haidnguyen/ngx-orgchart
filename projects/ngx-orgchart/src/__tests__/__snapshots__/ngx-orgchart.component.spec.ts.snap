// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgxOrgchartComponent should match snapshot 1`] = `
<div
  id="root0"
>
  
</div>
`;
